src/Models/Model.php

```diff
@@ -1,23 +1,25 @@
 <?php
-namespace SmartEmailing\v3\Models;
 
-use stdClass;
+declare(strict_types=1);
+
+namespace SmartEmailing\v3\Models;
 
 abstract class Model implements \JsonSerializable
 {
     /**
      * Copies the data from the JSON
-     * @param stdClass $json
+     *
+     * @param \stdClass $json
      *
      * @return static
      */
     public static function fromJSON($json)
     {
-        $item = new static();
+        $item = new static(); /** @phpstan-ignore-line */
```

Comment:
nedáme __construct v modelu jako final a tím pádem to bude správně?

Reply:
To bysme mohli

Reply:
Tak nemohli. Některé Modely redefinují konstruktor.

Reply:
Tak to pak ta funjkce nemůže fungovat správně. Kouknul jsem užití a nikdy fromJson se nepřebijí.

Co udělat toto:

- tam kde se funkce fromJson používá tak udělat novou do které dáme: final construct + fromJson.
- ta bude extendovat Model
- bude se jmenovat třeba jako `FromJsonModel`
- tak kde funkce byla použita tak použijeme místo třídy `Model` použijeme `FromJsonModel`

Hm? :)